### ISSUE TITLE
Simple .txt or .csv downloads of the datasets (especially the small ones)

### ISSUE THREAD

**Issue by PaulC61** [polaris-hub/polaris#209]
### Is your feature request related to a problem? Please describe.
The API you're developing is nice where you can load in and split the dataset for each benchmark. However, there is an issue of flexibility and control when people want to perform their own splitting regime for integration with previous studies. Especially in the case of the (currently) smaller datasets (biogen/ADME) where LOO-CV is really needed. 

### Describe the solution you'd like
It might not be a very "techy" way of doing things - but having the option to just download the datasets in a JSON/hdf5/.txt/.csv/**_simple_** format would be a nice feature to get up and running with integrating the curated datasets into existing workflows without a fuss. - Then the flexibility and control is up to _**us**_

**Comment by cwognum** [comment]
Hey @PaulC61, thanks for the kind words and thank you for the feedback!

To answer your question first, you can download all data associated with a dataset using 
```py
dataset.to_json(destination=...)
```
_(Side note: We should probably update the name of that function... Something like `download` or `save` seems more clear)_

As described in [the docs](https://polaris-hub.github.io/polaris/stable/api/dataset.html#polaris.dataset.Dataset.to_json), this will lead to the download of multiple files. In _most cases_, you will likely be interested in solely the `table.parquet` file, which is the Pandas DataFrame that holds the actual data. FYI - You can also access this same table using `dataset.table`.

Now having said that, we have something called "pointer columns" (see the [docs](https://polaris-hub.github.io/polaris/stable/tutorials/dataset_zarr.html#pointer-columns)) and are working on a new dataset implementation (to be announced soon!) that builds on that. This new implementation will actually forego the need of the `table` attribute in favor of a [Zarr](https://zarr.readthedocs.io/)-only implementation. These cases may make it more difficult to do what you're interested in!

So why do we do so? We have good reason to! We are purposefully offering up some flexibility in favor of standardization and easy of use.
- Standardization: If everyone were to have easy access to the raw dataset, we have little control over how people use the data. Different people may preprocess the data differently, leading to incomparable results. 
- Ease of use: We are trying to build a universal data format for drug discovery, so that different subcommunities (e.g. phenomics, small-molecules, ...) don't have to reinvent the wheel for data wrangling, but can all use the same Polaris API. Some more context [in this PyData talk](https://www.youtube.com/watch?v=YZDfD9D7mtE).

Does that make sense?

A final word: If you're interested in LOO-CV, you may be interested in our paper on method comparison which is coming out very soon! See [here](https://polarishub.io/guidelines/small-molecules) for more context.

**Comment by j-adamczyk** [comment]
This is really non-obvious. I absolutely was looking for `download` or `save` method! `dataset.table` for https://polarishub.io/datasets/asap-discovery/antiviral-admet-2025-unblinded throws an error `AttributeError: 'DatasetV2' object has no attribute 'table'`.

I don't care about API, about the library, I just want the CSV/Parquet/whatever file. The current interface is cumbersome, in fact exactly contrary to "ease of use". You don't and can't control how people use the data. I agree with providing recommendations, maybe even printing something if necessary in the code, but I really dislike using any frameworks for benchmarking, opaque libraries, "standardized" pipelines etc. I'm sure I am not the only one.

**Comment by cwognum** [comment]
Hi @j-adamczyk , thank you for the feedback! 

I appreciate that what we're building is not meant for everyone. We're on the very specific mission to change how people benchmark their methods in AI for drug discovery, to close the gap between perceived progress and real-world impact. We believe this mission requires us to reduce flexibility, in favour of standardization.

Luckily, you don't have to use Polaris if you disagree with our approach. Other solutions exist that give you all the flexibility you could ever wish for. In your specific case, you can for example also access the "raw" ASAP Antiviral data using their Zenodo link: https://zenodo.org/records/15582067

**Comment by cwognum** [comment]
If you're open to learn more about why we think this is important, please give one of our publications a read: 
- [A call for an industry-led initiative to critically assess machine learning for real-world drug discovery](https://www.nature.com/articles/s42256-024-00911-w)
- [Practically Significant Method Comparison Protocols for Machine Learning in Small Molecule Drug Discovery](https://pubs.acs.org/doi/10.1021/acs.jcim.5c01609)

(More coming!)

**Comment by j-adamczyk** [comment]
I agree with the approach and your papers (good work!), I just disagree with the API. Thanks for the Zenodo link! It's not linked on the dataset page. Including the dedicated field for publication link / original data hosting / other such thing would be pretty useful IMO.

Well, maybe just one caveat that I don't agree with using NHST hypothesis testing due to a range of papers on that (e.g. https://jmlr.org/papers/v7/demsar06a.html, https://www.jmlr.org/papers/v18/16-305.html, https://www.site.uottawa.ca/ICML08WS/papers/J_Demsar.pdf), and I prefer using Bayesian statistical tests with posterior distribution (e.g. https://www.jmlr.org/papers/v18/16-305.html, https://link.springer.com/article/10.1007/s10994-017-5641-9, https://www.jmlr.org/papers/v24/22-0907.html, https://arxiv.org/abs/2508.06199; I'm the co-author of the last one). But this is probably outside the scope of GitHub issue.

**Comment by cwognum** [comment]
> Including the dedicated field for publication link / original data hosting / other such thing would be pretty useful IMO.

Good feedback! Thanks! 

Note that we - the team behind _the platform_ - don't own the datasets or benchmarks. We work with partners, like ASAP, to upload their artifacts and then give them control. It's our partners that typically write the READMEs, for example. Nevertheless, it's good feedback and it's something I will share with our partners.

> I agree with the approach and your papers (good work!), I just disagree with the API. 

In my view, the API is in service of the approach. What would you do differently, beyond providing a link to the raw data? 

The issue with providing data in a raw format is that we lose control. People may report benchmarking results on top of slightly different versions of the same dataset (e.g. different filters or different preprocessing), making results incomparable. We currently see this all the time in the scientific literature. Our goal here is to become a reliable source of truth.

> I don't agree with using NHST hypothesis testing

Would love to learn more, but it's indeed out of scope for this issue. If you're interested, I would welcome you to open a discussion in https://github.com/polaris-hub/polaris-method-comparison/discussions . Then my co-author that has more of a background in statistics can chime in as well.

**Comment by j-adamczyk** [comment]
Just renaming the `.to_json()` method to `.download()` or `.save_local_copy()` or something like that would be useful. Or providing a separate method for that. This may be a niche consideration, but with bad/unstable internet, when on train/plane etc. (quite common at least for me), the current approach is really inconvenient, and local copy is much more reliable. I can do preprocessing or filters anyway with the current streamed version, right?

**Comment by cwognum** [comment]
Understood! Good feedback! I'll bring it back to the team. Thanks again!

>  I can do preprocessing or filters anyway with the current streamed version, right?

Yes. If you really wanted to, you can work your way around it. But as you've noticed, the API design discourages this. This is especially noticeable when you start using Polaris datasets in Polaris benchmarks.
